File: WoofAndroid/presentation/profile/src/main/kotlin/com/perrystreet/woof/presentation/profile/mapper/DogDomainToProfilePageUIModelMapper.kt
package com.perrystreet.woof.presentation.profile.mapper

import com.perrystreet.woof.models.dog.Dog
import com.perrystreet.woof.presentation.profile.uimodel.ProfilePageUIModel
import org.koin.core.annotation.Factory

@Factory
class DogDomainToProfilePageUIModelMapper {
    operator fun invoke(dog: Dog): ProfilePageUIModel = ProfilePageUIModel(
        id = dog.id,
        name = dog.name,
        photoUrl = dog.photoUrl,
        domain = dog,
    )
}

File: WoofIOS/SwiftPackages/Presentation/Profile/Sources/PresentationProfile/Mapper/DogDomainToProfilePageUIModelMapper.swift
import DI
import Models

@Factory
public final class DogDomainToProfilePageUIModelMapper {
    public func callAsFunction(dog: Dog) -> ProfilePageUIModel {
        ProfilePageUIModel(
            id: dog.id,
            name: dog.name,
            photoUrl: dog.photoUrl,
            domain: dog
        )
    }
}
